libbpf-tools/wakeuptime.bpf.c

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-3.0
+// Copyright (c) 2022 Nicolas Sterchele
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include "wakeuptime.h"
+#include "maps.bpf.h"
+
+#define PF_KTHREAD		0x00200000	/* kernel thread */
+
+const volatile pid_t targ_pid = 0;
+const volatile __u64 max_block_ns = -1;
+const volatile __u64 min_block_ns = 1;
+const volatile bool user_threads_only = false;
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct key_t);
+	__type(value, u64);
+} counts SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u32);
+	__type(value, u64);
+} start SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_STACK_TRACE);
+	__uint(key_size, sizeof(u32));
+} stackmap SEC(".maps");
+
+static __always_inline
+int offcpu_sched_switch(struct task_struct *prev) {
+		u64 pid_tgid = bpf_get_current_pid_tgid();
```

Comment:
* You don't need `__always_inline`
* `{` on a single line
* Use one tab for indention

Reply:
fixed in 8efb198